stdpopsim/slim_engine.py

```diff
@@ -948,17 +948,54 @@ def fix_time(event):
     condition_on_allele_frequency = []
     op_id = stdpopsim.ext.ConditionOnAlleleFrequency.op_id
     slim_mutation_ids = []
-    for x in _dfe_to_mtypes(contig).values():
+    slim_mutation_dfe = []
+    for i, x in _dfe_to_mtypes(contig).items():
         slim_mutation_ids.extend([u[0] for u in x])
+        slim_mutation_dfe.extend([i for u in x])
     for ee in extended_events:
+        if hasattr(ee, "mutation_id"):
+            mut_id = getattr(ee, "mutation_id")
+            cls_name = ee.__class__.__name__
+            mtype_idx = [
+                j
+                for j, i in enumerate(slim_mutation_dfe)
+                if contig.dfe_list[i].id == mut_id
+            ]
+            n_mtypes = len(mtype_idx)
+            # TODO: are these checks too restrictive? For example, these will
+            # prevent using DrawMutation with an arbitrary MutationType in an
+            # arbitary DFE.
+            if n_mtypes != 1:
+                raise ValueError(
+                    f"{cls_name} event with mutation_id '{mut_id}' must "
+                    f"map to a single MutationType instance on {contig}, "
+                    f"but has {n_mtypes} associated MutationType instances."
```

Comment:
I think this could be said more clearly? Something like "the mutation id associated with this event must be unique, but there are X other mutation types on this contig with the same mutation id."?